include/sparrow/layout.hpp

```diff
@@ -108,8 +111,9 @@ namespace sparrow
         using const_iterator = primitive_layout_iterator<T, true>;
```

Comment:
this should be: `using value_iterator = primitive_layout_value_iterator<T, false>` and equivalent for the const iterator.

`iterator` and `const_iterator` should be the alias to the iterators over the reference proxies. These iterators should be named `primitive_layout_iterator`